src/common/core/metacore.js

```diff
@@ -190,12 +200,20 @@ define([
                     break;
                 case 'integer':
                     if (!isNaN(parseInt(value)) && parseFloat(value) === parseInt(value)) {
-                        return true;
+                        if ((!meta.min || (meta.min && value >= meta.min)) &&
+                            (!meta.max || (meta.max && value <= meta.max))) {
```

Comment:
What if meta.min or meta.max is zero?
And why the extra check in the or clause?